backend/corpora/lambdas/api/v1/dataset.py

```diff
@@ -0,0 +1,39 @@
+from flask import make_response, jsonify
+
+from ....common.entities import Dataset
+from ....common.utils.db_utils import db_session
+from ....common.utils.exceptions import NotFoundHTTPException, ServerErrorHTTPException
+
+
+@db_session
+def post_dataset_asset(dataset_uuid: str, asset_uuid: str):
+
+    # retrieve the dataset
+    dataset = Dataset.get(dataset_uuid)
+    if not dataset:
+        raise NotFoundHTTPException(f"'dataset/{dataset_uuid}' not found.")
+
+    # retrieve the artifact
+    asset = dataset.get_asset(asset_uuid)
+    if not asset:
+        raise NotFoundHTTPException(f"'dataset/{dataset_uuid}/asset/{asset_uuid}' not found.")
+
+    # Retrieve S3 metadata
+    file_size = asset.get_file_size()
+    if not file_size:
+        raise NotFoundHTTPException(f"'dataset/{dataset_uuid}/asset/{asset_uuid}' not found.")
```

Comment:
Is this really a 404? It seems like more of an internal error if we have the asset in the db but we can't HEAD it.

Reply:
So if this happens, it likely means the the dataset_asset.s3_uri does not reference a valid s3 location in our bucket. This could happen if the underlying object was deleted after the asset was created, or the s3_uri is wrong during ingestion. We should add a check to make sure the resource exists before ingesting into the database which mean the only other possible reason is the object was deleted somehow.